gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/IdentityProvidersResource.java

```diff
@@ -133,14 +133,44 @@ public IdentityProviderResource getIdentityProviderResource() {
         return resourceContext.getResource(IdentityProviderResource.class);
     }
 
-    private IdentityProvider filterIdentityProviderInfos(IdentityProvider identityProvider) {
-        IdentityProvider filteredIdentityProvider = new IdentityProvider();
-        filteredIdentityProvider.setId(identityProvider.getId());
-        filteredIdentityProvider.setName(identityProvider.getName());
-        filteredIdentityProvider.setType(identityProvider.getType());
-        filteredIdentityProvider.setExternal(identityProvider.isExternal());
-        filteredIdentityProvider.setSystem(identityProvider.isSystem());
-        filteredIdentityProvider.setExternal(identityProvider.isExternal());
-        return filteredIdentityProvider;
+    private FilteredIdentityProviderInfo filterIdentityProviderInfos(IdentityProvider identityProvider) {
+        return new FilteredIdentityProviderInfo(identityProvider.getId(),
+                identityProvider.getName(), identityProvider.getType(), identityProvider.isSystem(), identityProvider.isExternal());
+    }
+
+    private static class FilteredIdentityProviderInfo {
```

Comment:
Maybe create this class inside `io.gravitee.am.model` ?

Reply:
Intentionally did not create this in `io.gravitee.am.model` folder as class as this class is a subset of the original model. Let's see what @remisultan suggest :)

Reply:
I believe the intent of the class is only to be used at Resource level. 

I think the best compromise is to have it in a separate class.
Not necessarily in `io.gravitee.am.model` but not as a private static class